### PR TITLE
Converted root package to main package

### PR DESCRIPTION
**Pull request by deneonet**
https://github.com/alecthomas/go_serialization_benchmarks/issues/162

### CODE REVIEW

**Review by alecthomas — APPROVED** [review]

File: benchmark.go [hunk 1]
```diff
@@ -78,6 +74,12 @@ func generateReport() error {
 		}
 	}
 
+	if !generateReport {
+		return nil
+	}
+
+	fmt.Printf("\nSaved report to report/data.js !\n\n")
```
Comment (matheusd) [inline benchmark.go:81]: At this point in the execution you haven't saved it yet.

**Review by matheusd — COMMENTED** [review]
This currently loses the ability to use the `-benchtime` flag, which means 

1- CI runs will take _much_ longer now (because we can't reduce the bench time target)
2- No way to improve accuracy by increasing the bench time target

I don't think there's a way to internally reach out to modify the benchtime arg (unless there's a very clever way) so I'd suggest still allowing running the benchmarks via `go test`, at least to improve the CI situation.

**Comment by deneonet** [comment]
Should be good now.

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -18,4 +18,4 @@ jobs:
         with:
           go-version: ${{ matrix.go }}
       - name: Verify benchmarks validate
-        run: go run . --validate
+        run: go test -v -bench . -run bench -benchtime 1ms
```
Comment (matheusd) [inline .github/workflows/go.yml:21]: This is not validating the benchmarks (see the job output)

https://github.com/alecthomas/go_serialization_benchmarks/actions/runs/9614218282/job/26518544178?pr=165

**Pull request merged**